adapters/Pangle/PangleAdapter/Bidding/GADPangleRTBNativeRenderer.m

```diff
@@ -180,13 +198,6 @@ - (BOOL)handlesUserImpressions {
   return YES;
 }
 
-- (GADNativeAdImage *)imageWithUrlString:(NSString *)urlString {
-  NSURL *url = [NSURL URLWithString:urlString];
-  NSData *data = [NSData dataWithContentsOfURL:url];
-  UIImage *image = [UIImage imageWithData:data];
-  return [[GADNativeAdImage alloc] initWithImage:image];
-}
-
```

Comment:
Can we do the below instead?
```
- (GADNativeAdImage *)loadIconWithUrlString:(NSString *)urlString {
  NSURL *url = [NSURL URLWithString:urlString];
  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
    NSData *data = [NSData dataWithContentsOfURL:url];
    dispatch_async(dispatch_get_main_queue(), ^{
      GADNativeAdImage *image = [[GADNativeAdImage alloc] initWithImage:[UIImage imageWithData:data]];
    });
    _icon = image;
  });
}

Reply:
`_icon = image` will report an error because the image is not within the scope of the current position. And this method will cause a delay in getting the icon.

Reply:
In the snippet, the `_icon = nil` obtained when invoke `_loadCompletionHandler`